bskyweb/cmd/bskyweb/server.go

```diff
@@ -454,6 +454,10 @@ func (srv *Server) WebPost(c echo.Context) error {
 				thumbUrls = append(thumbUrls, postView.Embed.EmbedRecordWithMedia_View.Media.EmbedImages_View.Images[i].Thumb)
 			}
 			data["imgThumbUrls"] = thumbUrls
+		} else if postView.Embed.EmbedVideo_View != nil {
+			data["width"] = postView.Embed.EmbedVideo_View.AspectRatio.Width
+			data["height"] = postView.Embed.EmbedVideo_View.AspectRatio.Height
+			data["videoUrl"] = fmt.Sprintf("https://%s/xrpc/com.atproto.sync.getBlob?did=%s&cid=%s", srv.xrpcc.Host, did, postView.Embed.EmbedVideo_View.Cid)
```

Comment:
- `xrpcc` is instantiated with the AppView URL, and AppView does not store raw blobs.
- We'd have to resolve the actual PDS that the user actually resides in
- This can incur unexpected bandwidth for third-party PDSes, especially when videos in Bluesky app itself is taken care of by a CDN.

Reply:
video urls can technically be any random object of your desire, for example youtube provides `<meta property="og:video:url" content="https://www.youtube.com/embed/kq39g2vSsKo">`. However most sites will have a whitelist of what providers are allowed to use an html embed, and all others are forced to use a well-known video container/encoding. If there's no such asset available and bluesky relies on m3u8, and this appview doesn't provide a whitelisted player embed like youtube does, a thumbnail is probably the best you can do.

Reply:
Since cbsky.app already handles video embeds properly, I wonder how it would react to a video stored on an external PDS (or even a currently-offline PDS). I wonder if it would be considered "sanitary" to link to the blob directly like cbsky does, just fetching it from the PDS directly instead of going through the CDN.

Reply:
Yes, I've mentioned this:

> Problem is, this can incur unexpected egress for third-party PDSes, especially noticaeble as videos in Bluesky app itself is taken care of by the CDN. Might not matter much to PDSes hosted in datacenters, but it's a dealbreaker for PDSes in home servers.

Reply:
These should definitely use a CDN URL, not a direct link to blob on the PDS. If it needs to just be a thumbnail image instead of the actual video, that is too bad but fine.

cc: @ericvolp12 on thoughts on this specifically.

Reply:
We have thumbnails generated by the CDN already. We're not generating MP4s or WebM or anything like that right now, it might be doable in the future but right now we may be best off serving a thumbnail. We have thumbnails generated by the CDN already at:

```
https://video.bsky.app/watch/$DID/$CID/thumbnail.jpg
```

Serving from the CDN would be our best bet.

Reply:
In regards to Discord specifically, I'm not sure this is much of an issue. Discord only queries the video file the first time the content is embedded and caches the file in its own CDN for playback in the future.
Repeated postings of the URL, even after a post is deleted from BSKY, will still continue to embed in Discord and show the content. (though I believe there is an eventual expiry date for when Discord will try and grab the content again to be more up to date, but it's quite a long time if there is one).

If Discord was at least whitelisted for this embed method, I'm not sure there would be much issue with it?

Reply:
The problem is that if you can make an exclusion for Discord, then you'd have to facilitate this exclusion for everyone else, or even just spoof Discord's user agent.

Discord does have a media cache, but that's not guaranteed to be permanent either.